extensions/util/calculate-eol.js

```diff
@@ -0,0 +1,39 @@
+'use strict';
+
+module.exports = (releaseDate, eolMonths, warningWeeks, logger) => {
+  if (!releaseDate) {
+    logger.warn(`No release date provided. Make sure to set {page-release-date} in the antora.yml of the component.`);
+    return null;
+  }
+
+  const targetDate = new Date(releaseDate);
+  if (isNaN(targetDate)) {
+    logger.warn('Invalid release date format:', releaseDate);
+    return null;
+  }
+
+  // Calculate EoL date
+  const eolDate = new Date(targetDate);
+  eolDate.setMonth(eolDate.getMonth() + eolMonths);
+
+  // Calculate the threshold for warning (X weeks before EoL)
+  const weeksBeforeEOL = new Date(eolDate);
+  weeksBeforeEOL.setDate(weeksBeforeEOL.getDate() - warningWeeks * 7);
+
+  // Determine if the current date falls within warning or post-EoL
+  const today = new Date();
```

Comment:
Using this method without setting a timezone can lead to discrepancies depending on where the server is located. 

Consider doing something like this to make sure that it's always executed in the same timezone. 

```js
const timeZone = 'UTC'; 

const now = new Date();
const formattedDate = new Intl.DateTimeFormat('en-US', {
  timeZone,
  year: 'numeric',
  month: '2-digit',
  day: '2-digit',
  hour: '2-digit',
  minute: '2-digit',
  second: '2-digit',
}).format(now);
```